chainer/links/loss/negative_sampling.py

```diff
@@ -29,11 +31,12 @@ class NegativeSampling(link.Link):
 
     """
 
-    def __init__(self, in_size, counts, sample_size, power=0.75):
+    def __init__(self, in_size, counts, sample_size, power=0.75, dtype=None):
         super(NegativeSampling, self).__init__()
+        dtype = chainer.get_dtype(dtype)
         vocab_size = len(counts)
         self.sample_size = sample_size
-        power = numpy.float32(power)
+        power = dtype.type(power)
```

Comment:
`power.dtype` in the next line could be fixed, while it's not necessarily related to the PR.

Reply:
I fixed!

Reply:
Sorry.  I meant “... could be fixed, too”.

Reply:
Just as follows?
```Python
power = dtype.type(power)
p = numpy.array(counts, dtype)
```

Reply:
Yes!

Reply:
Thanks!!